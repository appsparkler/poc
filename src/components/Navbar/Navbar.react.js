import React, {Component} from 'react'
import {ThemeContext} from '../../context/ThemeContext.react'

class Navbar extends Component {
  static contextType = ThemeContext
  render() {
    console.log(this.context)
    return (
      <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
        <a className="navbar-brand" href="/">Context App</a>
        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          <ul className="navbar-nav mr-auto">
            <li className="nav-item active">
              <a className="nav-link" href="/home">Home</a>
            </li>
            <li className="nav-item">
              <a className="nav-link" href="/about">About</a>
            </li>
            <li className="nav-item">
              <a className="nav-link" href="/contact">Contact</a>
            </li>
          </ul>
        </div>
      </nav>
    )
  }
}

export default Navbar
