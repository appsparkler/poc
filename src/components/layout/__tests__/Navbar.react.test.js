import React from 'react'
import renderer from 'react-test-renderer'
import {BrowserRouter as Router} from 'react-router-dom'
import Navbar from '../Navbar.react'

it('should render the navbar correctly', () => {
  const component = renderer.create(
      <Router>
        <Navbar />
      </Router>
  )
  const tree = component.toJSON()
  expect.assertions(1)
  expect(tree).toMatchSnapshot()
})
