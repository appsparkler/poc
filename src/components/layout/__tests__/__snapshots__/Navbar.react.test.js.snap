// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the navbar correctly 1`] = `
<nav>
  <div
    className="nav-wrapper grey darken-3"
  >
    <div
      className="container"
    >
      <a
        className="brand-logo"
        href="/"
        onClick={[Function]}
      >
        MarioPlan
      </a>
      <ul
        className="right"
      >
        <li>
          <a
            href="/create"
            onClick={[Function]}
          >
            Create Project
          </a>
        </li>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Logout
          </a>
        </li>
        <li>
          <a
            className="btn btn-floating pink lighten-3"
            href="/"
            onClick={[Function]}
          >
            NN
          </a>
        </li>
      </ul>
      <ul
        className="right"
      >
        <li>
          <a
            href="/signin"
            onClick={[Function]}
          >
            Sign Up
          </a>
        </li>
        <li>
          <a
            href="/signup"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
